web/users.go

```diff
@@ -93,6 +94,10 @@ func (u *userAuthRoundtrip) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 		}
 	}
 
+	if c.LogUnauthorized {
+		level.Info(u.logger).Log("msg", "Unauthorized", "ip", r.RemoteAddr, "forwarded", r.Header.Get("X-Forwarded-For"))
```

Comment:
Maybe make the forwarded for header configurable?

Reply:
Do you mean *if* the header should be used as the (only?) output or *what* header should be printed?

Reply:
I was thinking something like this:
```yaml
log_unauthorized:
  remote_ip_header: "X-Forwarded-For"
```

This would allow users to customize which `X-Forwarded-For` header to use (I've seen some proxies/CDNs use non-standard headers :shrug:)

Reply:
Good idea, I think I will just log one field "ip" then and use a header for that if specified in the config